airbyte-lib/airbyte_lib/progress.py

```diff
@@ -0,0 +1,318 @@
+"""A simple progress bar for the command line and IPython notebooks."""
+from __future__ import annotations
+
+import datetime
+import math
+import time
+from contextlib import suppress
+from typing import cast
+
+from rich.errors import LiveError
+from rich.live import Live as RichLive
+from rich.markdown import Markdown as RichMarkdown
+
+
+try:
```

Comment:
I really like these small QOL checks, but could we also add an option for this like pinecone does?

https://github.com/pinecone-io/pinecone-python-client/blob/61c19ad5a6a70321db44189dd90e603703670869/pinecone/data/index.py#L154

Some environments don't handle this kind of using the console very well and it's nice to be able to mute it for a clean output in a more "production" setting.

Reply:
I'm definitely on board for this. The refactoring would be significant at this stage though, specifically because the progress messages are currently sent from a few different places in the codebase. While I'm not able to do so in this iteration, I definitely think this is worth doing and will probably get to it down the road.